File: src/main/java/problems/easy/MaximumSubarray/MaximumSubarray.java
package problems.easy.MaximumSubarray;

@SuppressWarnings("PMD")
class Solution {
    public int maxSubArray(int[] nums) {
        var currentMaximum = Integer.MIN_VALUE;
        var maximumAtTheEnd = 0;
        for (var i = 0; i < nums.length; i++) {
            maximumAtTheEnd += nums[i];
            currentMaximum = maximumAtTheEnd > currentMaximum ? maximumAtTheEnd : currentMaximum;
            maximumAtTheEnd = maximumAtTheEnd < 0 ? 0 : maximumAtTheEnd;
        }
        return currentMaximum;
    }
}

File: src/main/java/problems/easy/MaximumSubarray/MaximumSubarray.kt
package problems.easy.MaximumSubarray

class MaximumSubarray {

    fun maxSubArray(nums: IntArray): Int {
        var currentMaximum = Integer.MIN_VALUE
        var maximumAtTheEnd = 0
        for (i in 0..nums.size - 1) {
            maximumAtTheEnd += nums[i]
            currentMaximum = if (maximumAtTheEnd > currentMaximum) maximumAtTheEnd else currentMaximum
            maximumAtTheEnd = if (maximumAtTheEnd < 0) 0 else maximumAtTheEnd
        }
        return currentMaximum
    }
}
